server/src/main/java/org/jboss/as/server/deploymentoverlay/AffectedDeploymentOverlay.java

```diff
@@ -0,0 +1,212 @@
+        Set<String> links = new HashSet<>();
+        for(String serverGoupName : serverGoupNames) {
+            links.addAll(listLinks(context, PathAddress.pathAddress(
+                    PathElement.pathElement(SERVER_GROUP, serverGoupName),
+                    PathElement.pathElement(DEPLOYMENT_OVERLAY, overlay))));
+        }
+        return links;
+    }
+
+    public static Set<String> listLinks(OperationContext context, PathAddress deploymentsRootAddress) {
+        Resource overlayResource = context.readResourceFromRoot(deploymentsRootAddress);
+        if (overlayResource.hasChildren(DEPLOYMENT)) {
+            return overlayResource.getChildrenNames(DEPLOYMENT);
+        }
+        return Collections.emptySet();
+    }
+
+    public static void redeployLinks(OperationContext context, PathAddress deploymentsRootAddress, Set<String> runtimeNames) throws OperationFailedException {
+        Set<String> deploymentNames = listDeploymentNames(context, deploymentsRootAddress, runtimeNames.stream().map(wildcardExpr -> Pattern.compile(wildcardToJavaRegex(wildcardExpr))).collect(Collectors.toSet()));
+        for (String deploymentName : deploymentNames) {
+            PathAddress address = deploymentsRootAddress.append(DEPLOYMENT, deploymentName);
+            OperationStepHandler handler = context.getRootResourceRegistration().getOperationHandler(address, REDEPLOY);
+            ModelNode operation = addRedeployStep(address);
+            assert handler != null;
+            assert operation.isDefined();
+            context.addStep(operation, handler, OperationContext.Stage.MODEL);
+        }
+    }
+
+    public static void redeployLinksAndTransformOperationForDomain(OperationContext context, Set<String> serverGroupNames, Set<String> runtimeNames) throws OperationFailedException {
+        Set<String> deploymentNames = new HashSet<>();
+        for (String serverGoupName : serverGroupNames) {
+            deploymentNames.addAll(listDeploymentNames(context, PathAddress.pathAddress(PathElement.pathElement(SERVER_GROUP, serverGoupName)), runtimeNames.stream().map(wildcardExpr -> Pattern.compile(wildcardToJavaRegex(wildcardExpr))).collect(Collectors.toSet())));
+        }
+        //Remote part
+        Operations.CompositeOperationBuilder opBuilder = Operations.CompositeOperationBuilder.create();
+        deploymentNames.forEach((deploymentName) -> opBuilder.addStep(addRedeployStep(context.getCurrentAddress().getParent().append(DEPLOYMENT, deploymentName))));
+        // Add the domain op transformer
+        List<DomainOperationTransformer> transformers = context.getAttachment(OperationAttachments.SLAVE_SERVER_OPERATION_TRANSFORMERS);
+        if (transformers == null) {
+            context.attach(OperationAttachments.SLAVE_SERVER_OPERATION_TRANSFORMERS, transformers = new ArrayList<>());
+        }
+        final ModelNode slave = opBuilder.build().getOperation();
+        transformers.add(new OverlayOperationTransformer(slave));
+    }
```

Comment:
@ehsavoie This makes me very nervous. But, it's very clever! So I'll go for it. But, two conditions:

1) There needs to be a PR in full adding coverage of both the new op and the remove+param case to the mixed domain testsuite. This would be necessary even if you did something less clever though.

2) There have to be really good comments added in quite a few places here. 3 months from now everyone but you will look at this and have no clue. 3 years from now you'll have no clue either. ;) It's not doing standard stuff so people won't be able to reason from the typical ways transformers work and domain rollout works, so it needs to be explained in the code. I'll add some comments now pointing out places where I'd like to see explanatory comments.